From 66a0dc242f890af1a19eb70706b5986bbdb4e33a Mon Sep 17 00:00:00 2001
From: Martin FERRAND <martin.ferrand@ponts.org>
Date: Wed, 8 Nov 2017 19:36:27 +0100
Subject: [PATCH 2/3] First question.

---
 src/base/fldprp.f90 | 3 +++
 src/base/fldvar.f90 | 8 ++++++++
 2 files changed, 11 insertions(+)

diff --git a/src/base/fldprp.f90 b/src/base/fldprp.f90
index 657bf2a..c13372c 100644
--- a/src/base/fldprp.f90
+++ b/src/base/fldprp.f90
@@ -181,6 +181,9 @@ endif
 call add_property_field_1d('courant_number', 'CFL', icour)
 call add_property_field_1d('fourier_number', 'Fourier Number', ifour)
 
+call add_property_field_1d('heavyside', 'Heavyside', iflid)
+call add_property_field_1d('dirac', 'Dirac', iflid)
+
 ! Total pressure is stored in property field of index iprtot
 ! if the compressible module is not enabled (otherwise Ptot=P*).
 ! For groundwater flows, this field is the pressure head (h = H - z),
diff --git a/src/base/fldvar.f90 b/src/base/fldvar.f90
index 943a9e2..eb80d3b 100644
--- a/src/base/fldvar.f90
+++ b/src/base/fldvar.f90
@@ -212,6 +212,14 @@ if (ivofmt.ge.0) then
   call field_set_key_struct_var_cal_opt(ivarfl(ivolf2), vcopt)
 endif
 
+if (ilevst.eq.1) then
+  call add_model_scalar_field('level_set', 'Level Set', ivolf2)
+  call field_get_key_struct_var_cal_opt(ivarfl(isca(ivolf2)), vcopt)
+  vcopt%idiff = 0
+  vcopt%iwarni = 2
+  call field_set_key_struct_var_cal_opt(ivarfl(isca(ivolf2)), vcopt)
+endif
+
 ! --- Turbulence
 
 if (itytur.eq.2) then
-- 
2.1.4

